crates/client/block_production/src/finalize_execution_state.rs

```diff
@@ -0,0 +1,328 @@
+use std::collections::{hash_map, HashMap};
+
+use blockifier::{
+    blockifier::transaction_executor::{TransactionExecutor, BLOCK_STATE_ACCESS_ERR},
+    bouncer::BouncerWeights,
+    state::{cached_state::StateMaps, state_api::StateReader},
+    transaction::errors::TransactionExecutionError,
+};
+use mc_db::{db_block_id::DbBlockId, MadaraBackend};
+use mc_mempool::MempoolTransaction;
+use mp_convert::ToFelt;
+use mp_state_update::{
+    ContractStorageDiffItem, DeclaredClassItem, DeployedContractItem, NonceUpdate, ReplacedClassItem, StateDiff,
+    StorageEntry,
+};
+use starknet_api::core::{ClassHash, CompiledClassHash, ContractAddress, Nonce};
+
+use crate::{Error, VisitedSegments};
+
+#[derive(Debug, thiserror::Error)]
+#[error("Error converting state diff to state map")]
+pub struct StateDiffToStateMapError;
+
+pub fn state_diff_to_state_map(diff: StateDiff) -> Result<StateMaps, StateDiffToStateMapError> {
```

Comment:
Can each field value be assigned its own variable please. The inlining here is killing me!